ocpp/charge_point.py

```diff
@@ -356,10 +356,25 @@ async def _handle_call(self, msg):
             call_unique_id_required = "call_unique_id" in handler_signature.parameters
             # call_unique_id should be passed as kwarg only if is defined explicitly
             # in the handler signature
+            inject_response = getattr(handler, "_inject_response", False)
             if call_unique_id_required:
-                response = handler(**snake_case_payload, call_unique_id=msg.unique_id)
+                if inject_response:
+                    response = handler(
+                        **snake_case_payload,
+                        call_unique_id=msg.unique_id,
+                        on_response=response_payload,
+                    )
+                else:
+                    response = handler(
+                        **snake_case_payload, call_unique_id=msg.unique_id
+                    )
             else:
-                response = handler(**snake_case_payload)
+                if inject_response:
+                    response = handler(
+                        **snake_case_payload, on_response=response_payload
+                    )
+                else:
+                    response = handler(**snake_case_payload)
```

Comment:
Not so "elegant" to have 4 calls to handler(). My suggestion is to use keyword arguments instead, like I did in https://github.com/mobilityhouse/ocpp/issues/264#issuecomment-1910237896 .

Reply:
This approach was suggested by @tropxy (one of the authors of this repo) in this [comment](https://github.com/mobilityhouse/ocpp/issues/264#issuecomment-964425685)

Reply:
@OSkrk what @astrand is suggesting is something like this (right @astrand ?):

```python
if "call_unique_id" in handler_signature.parameters:
    snake_case_payload["call_unique_id"] = msg.unique_id
if  inject_response:
    snake_case_payload["on_response"] = response_payload

response = handler(**snake_case_payload)
```

May you try that and see if it works?

Note: I have used your current nomenclature, but if you change the code then "on_response" will be "call_response"